### ISSUE TITLE
Is the no-unnecessary-semicolons rule necessary?

### ISSUE THREAD

**Issue by chrismbarr** [Microsoft/tslint-microsoft-contrib#376]
The `no-unnecessary-semicolons` rule seems do do the same thing as the `semicolon` rule _([docs](https://palantir.github.io/tslint/rules/semicolon/))_ distributed with TSLint 5.x (and probably previous versions as well)

I have the following configuration:
```json
"semicolon": {
    "options": [ "always", "ignore-interfaces", "ignore-bound-class-methods" ]
},
"no-unnecessary-semicolons": true
```

If I have the code:
```ts
interface IFoo {
  bar: string;
}; //<-- semicolon not needed here
```
Both of these rules report the exact same violation at the same location, and the error messages are nearly identical.

```
ERROR: 112:22  semicolon                  Unnecessary semicolon
ERROR: 112:22  no-unnecessary-semicolons  unnecessary semi-colon
```

To me it seems like this rule offers no advantage or difference from what the one distributed with TSLint provides.  The `no-unnecessary-semicolons` rule can probably be deprecated or removed.

**Comment by j-f1** [comment]
I think `no-unnecessary-semicolons` looks for places where semicolons are being over used. For example:

```
foo;;
```

**Comment by chrismbarr** [comment]
I just tried this with your example and I get the same violation from both rules.
```ts
return "foo";;
```

```
C:/path/to/file.ts
ERROR: 46:35  semicolon                  Unnecessary semicolon
ERROR: 46:35  whitespace                 missing whitespace
ERROR: 46:35  no-unnecessary-semicolons  unnecessary semi-colon
```

**Comment by alebrozzoSP** [comment]
I am using Prettier to format my code and it adds the semicolon on class properties that are arrow functions. When the property takes a single line, TSLint throws no error, but if it takes more than one line, I get the `Unnecessary semicolon` message. Since Prettier is formatting on save, I had to change TSL rules from error to warnings.
Here is an example

```
    // no error
    emptyAdvancedSearchFilters = (): AdvancedSearch => ({ firstName: "", lastName: "", email: "" });
    // error
    emptyAdvancedSearchFilters = (): AdvancedSearch => ({
        firstName: "",
        lastName: "",
        email: "",
        phone: "",
        school: "",
        chabadHouse: "",
        city: "",
        searchOptions: [],
    });
```

**Comment by pavestru** [comment]
It seems that `no-unnecessary-semicolons` was removed (cannot find it in the [docs](https://palantir.github.io/tslint/rules/)).

For me using this rule was sufficient:
```
"semicolon": [true, "always", "ignore-bound-class-methods"]
```

**Comment by HamletDRC** [comment]
Yes, we removed no-unnecessary-semicolons from our package because tslint offered 'semicolon' which covered all our cases.
